src/utils/response/responseHandlers.ts

```diff
@@ -4,7 +4,13 @@ import { ApiResponse } from './apiResponse';
 export type RequestPromise<T> = Promise<AxiosResponse<ApiResponse<T>>>;
 
 export async function toApiResponse<T>(promise: RequestPromise<T>) {
-  const apiResponse = (await promise).data;
+  const resolved = await promise;
+
+  if (resolved === undefined) {
```

Comment:
is that actually what the promise resolves with? I don't think it's supposed to, it should either resolve with an `AxiosResponse` or reject, so there should be no need for this if here